### PR TITLE
Documentation updates

### LINKED ISSUE #1354: Documentation request - explain acronyms in 'choosing a solver'
The documentation has a section called [choosing a solver](https://www.cvxpy.org/tutorial/advanced/index.html#choosing-a-solver).
That includes a table with many acronyms. I can guess some of them, but others I don't know.

Can you please add an explanation for each acronym below the table? First spell it out, and then maybe a sentence explaining it. (e.g. 'mixed integer' sounds like it does *not* include floats, but it does.)

* LP - Linear Programming?
* QP - ?
* SOCP - ?
* SDP - semi-definite positive? What does that mean?
* EXP - "Here EXP refers to problems with exponential cone constraints. The exponential cone is defined as ... " (already included in docs)
* POW - "The POW column refers to problems with 3-dimensional power cone constraints. The 3D power cone is defined as" ... (already included)
* MIP - mixed integers, i.e. can use ints/bools as well as floats

### PR DESCRIPTION
**Pull request by Transurgeon**
## Description
Here is a list of minor changes that I have made:
- resolve #1354 
- adding sphinx extension: autosectionlabel for internal linking to header sections
- fixing broken DGP link in advanced features
- updating hyperlinks in FAQ
- deleting the community folder as the information is already included in contributing/

P.S please ignore the ``readthedocs.yaml`` file, I am using it to preview doc changes which you can view in this link:
https://cvxpy--37.org.readthedocs.build/en/37/ 
Once the changes are approved, I will remove it from the PR. 
Issue link (if applicable): #1354 

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 sphinx
+sphinxcontrib.jquery
```
Comment (phschiele) [inline doc/requirements.txt:2]: I find that only these two are actually required. We should remove the others.

File: doc/source/tutorial/advanced/index.rst [hunk 2]
```diff
@@ -479,13 +479,24 @@ The table below shows the types of problems the supported solvers can handle.
...
 (***) Multiprecision support is available on SDPA if the appropriate SDPA package is installed. With multiprecision support, SDPA can solve your problem with much smaller `epsilonDash` and/or `epsilonStar` parameters. These parameters must be manually adjusted to achieve the desired degree of precision. Please see the solver website for details. SDPA can also solve some ill-posed problems with multiprecision support.
 
-Here EXP refers to problems with exponential cone constraints. The exponential cone is defined as
+LP - Linear Programming refers to problems with a linear objective function and linear constraints.
+
+QP - Quadratic Programming refers to problems with a quadratic objective function and linear constraints.
+
+SOCP - Second Order Cone Programming refers to problems with second-order cone constraints. The second order cone is defined as
+
+    :math:`C_{n+1} = \{\begin{bmatrix} x \\ t \end{bmatrix} \mid x \in R^n , t \in R , \| x \|_2 \leq t\}`
```
Comment (phschiele) [inline doc/source/tutorial/advanced/index.rst:488]:
```suggestion
    :math:`\mathcal{C}_{n+1} = \left\{\begin{bmatrix} x \\ t \end{bmatrix} \mid x \in \mathbb{R}^n , t \in \mathbb{R} , \| x \|_2 \leq t\right\}`
```

**Comment by phschiele** [comment]
Just had a few comments, otherwise looks good! Thanks for the update!

File: doc/requirements.txt (hunk 1 above)
Comment (Transurgeon) [reply doc/requirements.txt:2]: done

File: doc/source/tutorial/advanced/index.rst [hunk 3]
```diff
@@ -479,13 +479,24 @@ The table below shows the types of problems the supported solvers can handle.
...
-Here EXP refers to problems with exponential cone constraints. The exponential cone is defined as
+LP - Linear Programming refers to problems with a linear objective function and linear constraints.
+
+QP - Quadratic Programming refers to problems with a quadratic objective function and linear constraints.
+
+SOCP - Second Order Cone Programming refers to problems with second-order cone constraints. The second order cone is defined as
+
+    :math:`\mathcal{C}_{n+1} = \left\{\begin{bmatrix} x \\ t \end{bmatrix} \mid x \in \mathbb{R}^n , t \in \mathbb{R} , \| x \|_2 \leq t\right\}`
+
+SDP - Semi-Definite Programming refers to problems with :ref:`semi-definite constraints <Semidefinite matrices>`.
```
Comment (SteveDiamond) [inline doc/source/tutorial/advanced/index.rst:490]: Semidefinite Programming (no hyphen)

File: doc/source/tutorial/advanced/index.rst [hunk 4]
```diff
@@ -479,13 +479,24 @@ The table below shows the types of problems the supported solvers can handle.
...
+
+EXP - refers to problems with exponential cone constraints. The exponential cone is defined as
 
     :math:`\{(x,y,z) \mid y > 0, y\exp(x/y) \leq z \} \cup \{ (x,y,z) \mid x \leq 0, y = 0, z \geq 0\}`.
 
 Most users will never specify cone constraints directly. Instead, cone constraints are added when CVXPY
```
Comment (SteveDiamond) [inline doc/source/tutorial/advanced/index.rst:496]: Remove this paragraph. Doesn't fit in as well with the new text.

File: doc/source/tutorial/advanced/index.rst [hunk 5]
```diff
@@ -479,13 +479,24 @@ The table below shows the types of problems the supported solvers can handle.
 
 (***) Multiprecision support is available on SDPA if the appropriate SDPA package is installed. With multiprecision support, SDPA can solve your problem with much smaller `epsilonDash` and/or `epsilonStar` parameters. These parameters must be manually adjusted to achieve the desired degree of precision. Please see the solver website for details. SDPA can also solve some ill-posed problems with multiprecision support.
 
-Here EXP refers to problems with exponential cone constraints. The exponential cone is defined as
+LP - Linear Programming refers to problems with a linear objective function and linear constraints.
+
+QP - Quadratic Programming refers to problems with a quadratic objective function and linear constraints.
+
+SOCP - Second Order Cone Programming refers to problems with second-order cone constraints. The second order cone is defined as
```
Comment (SteveDiamond) [inline doc/source/tutorial/advanced/index.rst:486]: Second-Order Cone Programming (with hyphen)

File: doc/source/tutorial/advanced/index.rst (hunk 3 above)
Comment (SteveDiamond) [reply doc/source/tutorial/advanced/index.rst:490]: Also ``semidefinite matrix constraints``

**Comment by SteveDiamond** [comment]
Nice job! I had a few minor nits but otherwise looks good!

File: doc/source/tutorial/advanced/index.rst (hunk 4 above)
Comment (Transurgeon) [reply doc/source/tutorial/advanced/index.rst:496]: done

File: doc/source/tutorial/advanced/index.rst [hunk 6]
```diff
@@ -483,19 +483,16 @@ LP - Linear Programming refers to problems with a linear objective function and
 
 QP - Quadratic Programming refers to problems with a quadratic objective function and linear constraints.
 
-SOCP - Second Order Cone Programming refers to problems with second-order cone constraints. The second order cone is defined as
+SOCP - Second-Order Cone Programming refers to problems with second-order cone constraints. The second order cone is defined as
```
Comment (SteveDiamond) [inline doc/source/tutorial/advanced/index.rst:486]: should be ``second-order cone is defined``

Comment (Transurgeon) [reply doc/source/tutorial/advanced/index.rst]: done

**Review by PTNobel — APPROVED** [review]

**Review by SteveDiamond — APPROVED** [review]

**Pull request merged**